arches/app/views/api.py

```diff
@@ -1706,3 +1707,242 @@ def get(self,request):
+                    "description": spatialview.description,
+                    "geometrynodeid": str(spatialview.geometrynode.pk),
+                    "ismixedgeometrytypes": spatialview.ismixedgeometrytypes,
+                    "language": spatialview.language.code,
+                    "attributenodes": spatialview.attributenodes,
+                    "isactive": spatialview.isactive
+                } for spatialview in permitted_spatialviews
+            ]
+            return JSONResponse(response_data)
+
+    def attribute_nodes_are_valid(self, geom_node=models.Node, attribute_nodeids=[]):
+        attribute_node_graphs = [n.graph_id for n in models.Node.objects.filter(nodeid__in=attribute_nodeids)]
+        # if any of the graphs in attribute_node_graphs != geom_node.graph return False
+        are_valid = not any(graph_id != geom_node.graph_id for graph_id in attribute_node_graphs)
+
+        return are_valid
+
+
+    def identifier_is_uuid(self, identifier):
+        uuid_pattern = re.compile(settings.UUID_REGEX, re.IGNORECASE)
+        return uuid_pattern.match(identifier)
+
+
+    @method_decorator(group_required("Application Administrator"))
+    def post(self, request, identifier=None):
+        lang = None
+        isactive = False
+        ismixedgeometrytypes = None
+        description = None
+
+        try:
+            json_data = json.loads(request.body.decode('utf-8'))
+        except ValueError:
+            return JSONErrorResponse(_("Invalid JSON"), status=400)
+
+        if json_data is not None:
+            if "slug" not in json_data:
+                return JSONErrorResponse(_("No slug or Geometry Nodeid provided"), status=400)
+            if "geometrynodeid" not in json_data:
+                return JSONErrorResponse(_("No Geometry Nodeid provided"), status=400)
+            else:
+                try:
+                    geom_node = models.Node.objects.get(nodeid=json_data["geometrynodeid"])
+                except ObjectDoesNotExist:
+                    return JSONErrorResponse(_("No Node exists for supplied geometrynodeid"), status=400)
+            if "language" in json_data:
+                try:
+                    lang = models.Language.objects.get(code=json_data["language"])
+                except ObjectDoesNotExist:
+                    return JSONErrorResponse(_("No Language exists for supplied language code"), status=400)
+            else:
+                graph_x_pubs = models.GraphXPublishedGraph.objects.filter(graph=geom_node.graph)
+                pubs = models.PublishedGraph.objects.filter(publication__in=graph_x_pubs)
+                pub_graph_with_default_lang = list(filter(lambda x: x.language.isdefault is True, pubs))
+                try:
+                    lang = pub_graph_with_default_lang[0].language
+                except:
+                    return JSONErrorResponse(_("No default Language available in your Arches instance."), status=400)
```

Comment:
Need to check current published graph for the language.

`pubs = PublishedGraph.objects.filter(publication=geom_node.graph.publication) `

If there is a language in json_data then check it is in the published graph list, and if not reject.

If no language is given then should we not just reject like `slug`?